### PR TITLE
feat(project): allow querying ControlAction by target property

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
This pull request adds the possibility to query for ControlActions with a given target. This is useful to be able to get all ControlActions that originate from a given EntryPoint. Which can be used by algorithms that use a polling processing mechanism.

```gql
query {
  ControlAction(
    target: { identifier: "d7a3b614-4c40-413f-99d6-c0da2c844963" }
  ) {
    __typename
    identifier
    description
    actionStatus
  }
}
``` 

**TODO:** 

- [ ] @wimklerkx to be discussed; `targetIdentifier: "id"` VS `target: { identifier: "id" }` 
- [ ] Create an abstract class for the GetQuery and GetControlActionByTargetQuery classes.

### CODE REVIEW

**Comment by wimklerkx** [comment]
Codacy looks like a fantastic reviewer, but I think it's better to not use it for this project as
- it does not agree with the 'standard' linter used for the project (semicolons etc)
- it might put off developers of TROMPA participants who want to contribute and see their work rejected by a bot that cannot be argued with

**Comment by wimklerkx** [comment]
On this: @wimklerkx to be discussed; targetIdentifier: "id" VS target: { identifier: "id" }

I think it's better to do this: targetIdentifier: "id"
As it is more straightforward for the API user and less code in general

This approach becomes necessary though when extra parameters would be needed to identify the right EntryPoint. But that's not the case here

**Comment by wimklerkx** [comment]
On this: Create an abstract class for the GetQuery and GetControlActionByTargetQuery classes.

Yes, at some point it becomes necessary to even create an abstract 'Query' class
So, feel free to improve :)

File: api/src/resolvers/query.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { info } from '../index'
 import { driver } from '../driver'
 import { retrieveNodeData, hydrateNodeSearchScore } from '../resolvers'
+import GetControlActionByTargetQuery from '../queries/GetControlActionByTargetQuery';
```
[the remark this answers is not in the document]
Comment (wimklerkx) [reply api/src/resolvers/query.js]: We use https://standardjs.com

So no semicolons, weird as that might initially feel

File: api/src/queries/GetControlActionByTargetQuery.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import StringHelper from '../helpers/StringHelper'
```
Comment (wimklerkx) [inline api/src/queries/GetControlActionByTargetQuery.js]: My code breaks on this spread operator: (Node 10.13)

/Users/wim/.nvm/versions/node/v10.13.0/bin/node -r babel-register /Users/wim/TROMPA/CE-api/api/src/index.js
/Users/wim/TROMPA/CE-api/api/node_modules/babel-core/lib/transformation/file/index.js:558
      throw err;
      ^

SyntaxError: /Users/wim/TROMPA/CE-api/api/src/queries/GetControlActionByTargetQuery.js: Unexpected token (23:34)
  21 | 
  22 |     // split the params into targetParams and baseParams
> 23 |     const { target: targetParams, ...baseParams } = this.params

Comment (ChristiaanScheermeijer) [reply api/src/queries/GetControlActionByTargetQuery.js]: Fixed by merging branch `feat/update-babel`

**Comment by ChristiaanScheermeijer** [comment]
The ControlAction query has been updated with `targetIdentifier` VS `target: { ... }`. The destructuring error is fixed by merging the new babel version.

> Codacy looks like a fantastic reviewer, but I think it's better to not use it for this project as
> 
> * it does not agree with the 'standard' linter used for the project (semicolons etc)
> * it might put off developers of TROMPA participants who want to contribute and see their work rejected by a bot that cannot be argued with

1) It hadn't been setup properly. I've submitted a PR which should fix all code style errors. Since standard JS was installed in the ./api directory, it was actually not working properly (specifically the precommit hook). In order to make it work for both; precommits and codacy, I had to switch to eslint with the standard config.
2) It's for everyone's best interest to force a code style. The choice of code style can be argued with though.

**Pull request merged**